src/databricks/labs/ucx/installer/workflows.py

```diff
@@ -609,6 +613,15 @@ def _job_clusters(self, names: set[str]):
             )
         return clusters
 
+    def _job_parse_logs_task(self, job_tasks: list[jobs.Task], workflow: str, remote_wheel: str) -> jobs.Task:
+        jobs_task = jobs.Task(
+            task_key=TASK_NAME_PARSE_LOGS,
+            job_cluster_key=Task.job_cluster,
+            # The task dependents on all previous tasks.
+            depends_on=[jobs.TaskDependency(task_key=task.task_key) for task in job_tasks],
```

Comment:
I removed the `jobs.RunIf.ALL_DONE` in f0b68e1. Because running the `parse_logs` with an `ALL_DONE` at the end of each workflow changes the workflow status. In our testing suite, the workflows that we expect to fail do not fail not more. In production, users get a less obvious hint when the job failed. (It will be succeeded with failures instead of a failure.)

IMO it makes sense to only run the `parse_logs` if all succeed. In the case a job fails, you won't be looking at the dashboard the failing job should have created. You will be looking at the job logs in the workflow UI. And maybe later using the CLI to fetch all failing records. What is your opinion?

Reply:
> Because running the parse_logs with an ALL_DONE at the end of each workflow changes the workflow status.

i see. 

>  In the case a job fails, you won't be looking at the dashboard the failing job should have created. You will be looking at the job logs in the workflow UI. And maybe later using the CLI to fetch all failing records. 

this is correct, when users are more tech-savy, but they are mostly not.

> What is your opinion?

what about the following flow:
1. run_if=ALL_DONE
2. get distinct ERROR messages
3. if there's more than zero ERROR messages, raise the exception with those, failing the job

This way:
1. we'll reveal hidden bugs from seemingly passing tasks
2. reveal warnings for passing tasks
3. won't change the external behaviour expectation about the workflow